cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ClusterCa.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.model;
+
+import io.fabric8.kubernetes.api.model.Secret;
+import io.strimzi.api.kafka.model.Kafka;
+import io.strimzi.certs.CertAndKey;
+import io.strimzi.certs.CertManager;
+import io.strimzi.certs.Subject;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+public class ClusterCa extends Ca {
+
+    // the Kubernetes service DNS domain is customizable on cluster creation but it's "cluster.local" by default
+    // there is no clean way to get it from a running application so we are passing it through an env var
+    public static final String KUBERNETES_SERVICE_DNS_DOMAIN =
+            System.getenv().getOrDefault("KUBERNETES_SERVICE_DNS_DOMAIN", "cluster.local");
+    private final String clusterName;
+    private Secret entityOperatorSecret;
+    private Secret topicOperatorSecret;
+
+    private Secret brokersSecret;
+    private Secret zkNodesSecret;
+
+    public ClusterCa(CertManager certManager, String clusterName, Secret caCertSecret, Secret caKeySecret) {
+        this(certManager, clusterName, caCertSecret, caKeySecret, 365, 30, true);
+    }
+
+    public ClusterCa(CertManager certManager,
+                     String clusterName,
+                     Secret clusterCaCert,
+                     Secret clusterCaKey,
+                     int validityDays,
+                     int renewalDays,
+                     boolean generateCa) {
+        super(certManager, AbstractModel.getClusterCaName(clusterName), clusterCaCert,
+                AbstractModel.getClusterCaKeyName(clusterName), clusterCaKey,
+                validityDays, renewalDays, generateCa);
+        this.clusterName = clusterName;
+    }
+
+    public void sort(List<Secret> secrets) {
```

Comment:
Why is this called `sort`? It doesn't seem to `sort` anything. It just decodes from list into fields.